src/sonarcloud/service.ts

```diff
@@ -0,0 +1,80 @@
+import fetch from "node-fetch"
+import { Config } from "../config"
+import { SonarCloudTokenCliProvider, SonarCloudTokenProvider } from "./token"
+import { SonarCloudProjectMetrics } from "./types"
+
+interface SonarCloudServiceProps {
+  config: Config
+  tokenProvider: SonarCloudTokenProvider
+}
+
+export class SonarCloudService {
+  private config: Config
+  private tokenProvider: SonarCloudTokenProvider
+
+  public constructor(props: SonarCloudServiceProps) {
+    this.config = props.config
+    this.tokenProvider = props.tokenProvider
+  }
+
```

Comment:
doc: Could be useful with function documentation that explains that this currently only returns coverage metrics